File: fix_line_endings.js
const fs = require('fs');
const path = require('path');

const root = process.cwd();
const excludeDirs = new Set(['.git', 'node_modules', 'dist']);
const textExtensions = new Set(['.js', '.css', '.html', '.json', '.md', '.yaml', '.yml', '.txt']);

function walk(dir) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      if (!excludeDirs.has(entry.name)) {
        walk(fullPath);
      }
      continue;
    }
    if (!entry.isFile()) {
      continue;
    }
    const ext = path.extname(entry.name).toLowerCase();
    if (!textExtensions.has(ext)) {
      continue;
    }
    const content = fs.readFileSync(fullPath, 'utf8');
    if (content.includes('\r\n')) {
      fs.writeFileSync(fullPath, content.replace(/\r\n/g, '\n'));
    }
  }
}

walk(root);

File: fix_line_endings.py
from pathlib import Path

root = Path('.')
exclude_dirs = {'.git', 'node_modules', 'dist'}
text_extensions = {'.js', '.css', '.html', '.json', '.md', '.yaml', '.yml', '.txt'}

for path in root.rglob('*'):
    if not path.is_file():
        continue
    if any(part in exclude_dirs for part in path.parts):
        continue
    if path.suffix.lower() not in text_extensions:
        continue
    try:
        text = path.read_text(encoding='utf-8')
    except Exception:
        continue
    if '\r\n' in text:
        path.write_text(text.replace('\r\n', '\n'), encoding='utf-8', newline='\n')
